src/key.js

```diff
@@ -623,6 +623,86 @@ Key.prototype.revoke = function() {
 };
 
 /**
+ * Signs the public key
+ * @param  {Array<module:key~Key>} privateKey decrypted private keys for signing
+ * @return {module:key~Key} new public key with new certificate signature
+ */
+Key.prototype.sign = function(privateKeys) {
+  var primaryUser, primaryKey, dataToSign, signingKeyPacket, signaturePacket, user, key;
+  if (this.isPrivate()) {
+    throw new Error('Only public key can be signed');
+  }
+  primaryUser = this.getPrimaryUser();
+  if (!primaryUser) {
+    throw new Error('Could not find primary user');
+  }
+  primaryKey = this.primaryKey;
+  dataToSign = {};
+  dataToSign.userid = primaryUser.user.userId || primaryUser.user.userAttribute;
+  dataToSign.key = primaryKey;
```

Comment:
`Key.prototype.signAllUsers` is well-looking option to set some day another user as primary and not collect again all trusted signatures.
About functionality located at User.prototype, we will have to also always pass primaryKey (of public key) to prototype methods, cause they need both to sign and verify.
If I understand correctly userIds cant be signed without related keyid, otherwise someone could put own primaryKey and self-signature to trusted public key.
About signing of other user keys, I think it is not neccesary cause they are bound with primaryKey by binding signatures